prow/githuboauth/githuboauth.go

```diff
@@ -291,7 +295,7 @@ func (ga *Agent) HandleRedirect(client OAuthClient, identifier AuthenticatedUser
 			ga.serverErrorAndPrint(w, "Save session", err)
 			return
 		}
-		user, err := identifier.LoginForRequester("oauth", token.AccessToken)
+		user, err := identifier.LoginForRequester("oauth", token.AccessToken, false)
```

Comment:
Why is this `false`? And more specifically, why is it ever `false`?

When prow uses a PAT, we will always use the `token.AccessToken`, regardless of the `usePAT` argument in `GitHubClientWithAccessToken` as `getToken` gets unconditionally overwritten in there.

When prow uses apps auth, we will now only use the supplied token when `usePAT` is true and otherwise fall back to Prows own access, resulting in the `user` here being the github app, which seems wrong.

To me, it is fundamentally unexpected that a client constructor that takes a token will under any circumstances construct a client that does not use the provided token. If you want to use the base auth, you should just call `flagutil.GitHubClient()` rather than `GitHubClientWithAccessToken()`.

Reply:
That is a good point, and I am no longer sure why I originally had this instance as `false`...something else must have been breaking it as it works fine now.

I have updated the `GitHubClientWithAccessToken` method to **always** set `AppId` to the empty string.